proposals/20240901-drop-support-for-syscall-enter-events.md

```diff
@@ -0,0 +1,353 @@
+Syscall stats (userspace-side):
+- [epoll_ctl__enter]: 55.002.263
+- [epoll_ctl__exit]: 55.002.263
+- [gettimeofday__exit]: 54.841.309
+- [gettimeofday__enter]: 54.841.309
+- [clock_gettime__enter]: 49.684.503
+- [clock_gettime__exit]: 49.684.503
+- [write__exit]: 11.022.850
+- [write__enter]: 11.022.850
+- [read__enter]: 11.012.631
+- [read__exit]: 11.012.631
+- [sendto__enter]: 11.000.016
+- [sendto__exit]: 11.000.016
+- [recvfrom__enter]: 11.000.015
+- [recvfrom__exit]: 11.000.015
+- [epoll_wait__exit]: 5.878.219
+- [epoll_wait__enter]: 5.878.219
+- [clock_nanosleep__exit]: 84.301
+- [clock_nanosleep__enter]: 84.300
+- [close__exit]: 7696
+- [close__enter]: 7696
+- [setsockopt__enter]: 2905
+- [setsockopt__exit]: 2905
+- [fcntl__enter]: 1371
+- [fcntl__exit]: 1371
+- [accept4__enter]: 749
+- [accept4__exit]: 749
+- [socket__enter]: 608
+- [socket__exit]: 608
+- [connect__enter]: 598
+- [connect__exit]: 598
+------------------------------------------------------------------
+
+PING_INLINE: 42084.00 requests per second, p50=0.615 msec                
+PING_MBULK: 41743.20 requests per second, p50=0.623 msec                
+SET: 42059.22 requests per second, p50=0.615 msec                
+GET: 42060.99 requests per second, p50=0.615 msec                
+INCR: 41704.89 requests per second, p50=0.623 msec                
+LPUSH: 41557.58 requests per second, p50=0.623 msec                
+RPUSH: 41951.59 requests per second, p50=0.623 msec                
+LPOP: 42301.18 requests per second, p50=0.615 msec                
+RPOP: 41630.24 requests per second, p50=0.623 msec                
+SADD: 41785.06 requests per second, p50=0.623 msec                
+HSET: 41618.11 requests per second, p50=0.623 msec 
+```
+
+* Here we see an instrumentation time of `146,5` s, **more or less splitted equally between `sys_enter` and `sys_exit`.** This is probably a best scenario because some of these syscalls are treated as "generic" by our probe so the instrumentation time of the enter and exit events should be almost the same, but we can easily understand that even in a not ideal scenario the `sys_enter` program has a great impact on our instrumentation overhead.
+* We set yet another decrease in Redis performance. This is due to 2 main factors:
+  * a greater kernel instrumentation time.
+  * a greater CPU request from scap-open that now needs to process 419 million events.
+* We send to Userspace `12.6` GB in `263` seconds
+
+## Required steps
+
+This activity involves several steps at different levels:
+
+* Update our three drivers to collect all parameters in the exit fillers and remove the enter ones.
+* Update scap-file management to convert all enter events generated over the years into new exit events.
+* Update our userspace engines to support only exit events.
+* Update the sinsp parsing and extraction logic to use only exit events.
+* Update all tests (drivers, libscap, unit, e2e) to use only exit events.
+* Update all documentation.
```

Comment:
This is a lot of work, and I mean *A LOT* of work. From where I stand, this is basically a rewrite of the parsing logic of sinsp and all 3 drivers. This is a ton of risk and I'm not convinced moving everything to exit events will have the huge impact you are expecting it to have. In my mind, for me to sign off on this, you'd need to have CPU and data savings of way above 50%, otherwise it's too much risk of having regressions.